packages/istanbul-lib-instrument/src/visitor.js

```diff
@@ -339,16 +339,16 @@ function blockProp(prop) {
     };
 }
 
-function makeParenthesizedExpression(path) {
-  var T = this.types;
-  if (path.node) {
-    path.replaceWith(T.parenthesizedExpression(path.node));
-  }
+function makeParenthesizedExpressionForNonIdentifier(path) {
+    var T = this.types;
+    if (path.node && !path.isIdentifier()) {
```

Comment:
what parse would this add parenthesis around?

Reply:
The parenthesization is introduced in #106 to support conditional superClass. Besides the conditional expression, I can think of a function case that also needs parenthesis to wrap the instrumented result, though it should be pretty rare and even theoretic only.
```js
class App extends ClassFactory({ configurable: false }, baseClass) {
}
```

As the superClass can be any left hand side expression, I exclude the parenthesis process when the superClass is simply an identifier, since it need not instruments.